File: app/src/main/java/io/connorwyatt/flashcards/views/directionalviewpager/PageChangeListener.java
package io.connorwyatt.flashcards.views.directionalviewpager;

import android.support.v4.view.ViewPager;
import io.connorwyatt.flashcards.adapters.FlashcardTestPagerAdapter;
import io.connorwyatt.flashcards.data.entities.Flashcard;

class PageChangeListener implements ViewPager.OnPageChangeListener {
    private DirectionalViewPager context;
    private boolean hasPageChanged = false;
    private int previousPage;
    private int currentPage;

    PageChangeListener(DirectionalViewPager context) {
        this.context = context;
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
    }

    @Override
    public void onPageSelected(int position) {
        hasPageChanged = true;
        previousPage = currentPage;
        currentPage = position;
    }

    @Override
    public void onPageScrollStateChanged(int state) {
        switch (state) {
            case ViewPager.SCROLL_STATE_IDLE:
                if (hasPageChanged) {
                    if (!context.getAllowLeftSwipe() && currentPage > previousPage) {
                        FlashcardTestPagerAdapter adapter = (FlashcardTestPagerAdapter) context.getAdapter();

                        if (adapter != null) {
                            Flashcard removedFlashcard = adapter.removeItem(previousPage);
                            context.setCurrentItem(previousPage, false);
                            hasPageChanged = false;
                            context.notifySkipped(removedFlashcard);
                        }
                    }
                }

                break;
        }
    }
}

File: app/src/main/java/io/connorwyatt/flashcards/views/directionalviewpager/PageChangeListener.kt
/*
 * Copyright (c) 2016-2017 Connor Wyatt <connorwyatt1@gmail.com>.
 *
 * This file can not be copied and/or distributed without the express permission of Connor Wyatt.
 */

package io.connorwyatt.flashcards.views.directionalviewpager

import android.support.v4.view.ViewPager
import io.connorwyatt.flashcards.adapters.FlashcardTestPagerAdapter

internal class PageChangeListener(private val context: DirectionalViewPager) : ViewPager.OnPageChangeListener
{
    private var hasPageChanged = false
    private var previousPage: Int = 0
    private var currentPage: Int = 0

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int)
    {
    }

    override fun onPageSelected(position: Int)
    {
        hasPageChanged = true
        previousPage = currentPage
        currentPage = position
    }

    override fun onPageScrollStateChanged(state: Int)
    {
        when (state)
        {
            ViewPager.SCROLL_STATE_IDLE ->
                if (hasPageChanged && !context.allowLeftSwipe && currentPage > previousPage)
                {
                    val adapter = context.adapter as FlashcardTestPagerAdapter

                    val removedFlashcard = adapter.removeItem(previousPage)
                    context.setCurrentItem(previousPage, false)
                    hasPageChanged = false
                    context.notifySkipped(removedFlashcard)
                }
        }
    }
}
